silverstripe/silverstripe-admin
# FIX Repalce mouseDown with keyDown event on treedropdown option button

## Description

Replaced unnecessary `mouseDown` event with `keyDown` event to prevent double invoking `handleNavigate` method.
New behat test cases were added to the [`silverstripe\cms`](https://github.com/silverstripe/silverstripe-cms/pull/2931) module.  

## Manual testing steps


Reproduction steps
Will require that you have a parent page with at least two levels of child pages
- Create Page with few nested child and grandchild
- Go to any TinyMCE content block on the site
- Select the Insert link and then select Page on this site
- Select the Select a page dropdown
- Select the arrow to view a pages child pages on option for new created Page, if it has more than one child, then you should see child Page and not grandchild Page.

## Issues

- https://github.com/silverstripe/silverstripe-cms/issues/2930

## Pull request checklist

    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
